python/tskit/trees.py

```diff
@@ -3822,6 +3853,7 @@ def __init__(self, tree):
         self.tree = tree
         self.more_trees = True
         self.forward = True
+        self.no_skip_empty = True if tree.skip_empty is None else not tree.skip_empty
```

Comment:
More natural to make ``skip_empty`` a parameter of the TreeIterator class, so we call it like:
```
return TreeIterator(tree, skip_empty=skip_empty)
```